docs/source/custom_models.mdx

```diff
@@ -0,0 +1,153 @@
+<!--Copyright 2020 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Sharing custom models
+
+The 🤗 Transformers library is designed to be easily extensible. Every model is fully coded in a given subfolder
+of the repository with no abstraction, so you can easily copy a modeling file and tweak it to your needs.
+
+Once you are happy with those tweaks and trained a model you want to share with the community, there are simple steps
+to push on the Model Hub not only the weights of your model, but also the code it relies on, so that anyone in the
+community can use it, even if it's not present in the 🤗 Transformers library.
+
+This also applies to configurations and tokenizers (support for feature extractors and processors is coming soon).
+
+## Sending the code to the Hub
+
+First, make sure your model is fully defined in a `.py` file. It can rely on relative imports to some other files as
+long as all the files are in the same directory (we don't support submodules for this feature yet). For instance,
+let's say you have a `modeling.py` file and a `configuration.py` file in a folder of the current working directory
+named `awesome_model`, and that the modeling file defines an `AwesomeModel`, the configuration file a `AwesomeConfig`.
+
+```
+.
+└── awesome_model
+    ├── __init__.py
+    ├── configuration.py
+    └── modeling.py
```

Comment:
shouldn't it be called `awesome_modeling.py` in that case?

Reply:
Why? It's user-defined, they can do what they want.